core/blocktree/blocktree.go

```diff
@@ -32,30 +33,31 @@ type Hash = common.Hash
 
 // BlockTree represents the current state with all possible blocks
 type BlockTree struct {
-	head            *node
+	head            *Node
 	leaves          leafMap
-	finalizedBlocks []*node
+	finalizedBlocks []*Node
 	Db              *polkadb.BlockDB
 }
 
 // NewBlockTreeFromGenesis initializes a blocktree with a genesis block.
 func NewBlockTreeFromGenesis(genesis types.Block, db *polkadb.BlockDB) *BlockTree {
-	head := &node{
-		hash:     genesis.Header.Hash,
-		number:   genesis.Header.Number,
-		parent:   nil,
-		children: []*node{},
-		depth:    big.NewInt(0),
+	head := &Node{
+		Hash:     		genesis.Header.Hash,
+		Number:   		genesis.Header.Number,
+		parent:   		nil,
+		children: 		[]*Node{},
+		depth:    		big.NewInt(0),
+		ArrivalTime: 	uint64(time.Now().UnixNano()), // set arrival time of genesis in nanoseconds since unix epoch
```

Comment:
wondering if the arrival time should be documented here (when block is put into blocktree) or when the node is started up? and for other nodes, when they arrive from the network?